src/coreclr/jit/codegenarm64.cpp

```diff
@@ -3022,6 +3022,18 @@ void CodeGen::genCodeForNegNot(GenTree* tree)
     regNumber   targetReg = tree->GetRegNum();
     instruction ins       = genGetInsForOper(tree->OperGet(), targetType);
 
+    if ((tree->gtFlags & GTF_SET_FLAGS) != 0)
```

Comment:
There is a function `GenTree::gtSetFlags`, would be nice to use this if possible.

It looks like it has a couple additional checks on the node type and such that might need to be taken into consideration.

Reply:
I remember seeing `gtSetFlags` and `gtRequestSetFlags` - currently, there is precedent of not using these in codegenarm64.cpp and in lowerarmarch.cpp.

I think because of this precedent, we should keep it the way it is and have a separate PR go in and change all the cases where we do these kinds of checks to use `gtSetFlags` and `gtRequestSetFlags`.